src/Cli/PhpCommands.php

```diff
@@ -32,6 +32,104 @@ protected function phpIsEOL($work_dir, $version)
+
+        $cos_php = WorkingCopy::clone($url, $work_dir, $api);
+        $cos_php
+            ->addFork($fork)
+            ->setLogger($this->logger);
+        $this->logger->notice("Check out {project} to {path}.", ['project' => $cos_php->projectWithOrg(), 'path' => $work_dir]);
+        $cos_php
+            ->switchBranch('master')
+            ->pull('origin', 'master');
+
+        $versions_file_path = "$work_dir/PHP_VERSIONS";
+        $version_file_contents = file_get_contents($versions_file_path);
+        $updated_versions = [];
+
+        $versions = $this->getCosPhpVersions($version_file_contents);
+
+        foreach ($versions as $version => $prefix) {
+            $next_version = $this->nextVersionThatExists($version);
+            if (!$next_version) {
+                $urlTemplate = $this->getConfig()->get('php-net.download-url');
+                throw new \Exception("Could not find current php $version on php downloads server. Using url: $urlTemplate. Check configuration and network.");
+            }
+
+            if ($next_version != $version) {
+                $this->say("$next_version is available.");
+                $version_file_contents = preg_replace("#$prefix$version#", "$prefix$next_version", $version_file_contents);
+                $updated_versions[] = $next_version;
+            } else {
+                $this->say("$version is the most recent version on COS");
+            }
+        }
+
+        if (empty($updated_versions)) {
+            return;
+        }
+
+        $all_updated_versions = $this->prettyImplode(', ', ' and ', array_map(function ($v) {
+            return "php-$v";
+        }, $updated_versions));
+        $preamble = $this->preamble();
+        $message = "{$preamble}{$all_updated_versions}";
+
+        // Determine if there are any PRs already open that we should
+        // close. There should be no more than one. If its contents are the
+        // same, then we should abort rather than create the same PR again.
+        // If the cnotents are different, then we'll make a new PR and close
+        // this one.
+        $prs = $api->matchingPRs($cos_php->projectWithOrg(), $preamble, '');
+        if (in_array($message, $prs->titles())) {
+            $this->logger->notice("There is an existing pull request for this update; nothing else to do.");
+            return;
+        }
+
+        file_put_contents($versions_file_path, $version_file_contents);
+
+        // Create a new pull request
+        $branch = $this->branchPrefix() . implode('-', $updated_versions);
+        $this->logger->notice('Using {branch}', ['branch' => $branch]);
+        $cos_php
+            ->createBranch($branch, 'master', true)
+            ->add('PHP_VERSIONS')
+            ->commit($message)
+            ->push()
+            ->pr($message);
+
+        // Once we create a new PR, we can close the existing PRs.
+        $api->prClose($cos_php->org(), $cos_php->project(), $prs);
```

Comment:
Would be nice to link to new PR when closing old ones if that's straightforward.

Reply:
That's a good suggestion, but there are a number of places where we close old PRs in favor of a new one. I think I'll leave this as follow-on work, so I can do it consistently without putting our sprint goals at risk.

Reply:
Created [OTTO-334](https://getpantheon.atlassian.net/browse/OTTO-334)